internal/docs/design.md

```diff
@@ -141,6 +141,94 @@ canonical example is `gcpkms` and `awskms`.
 [cascading failure]:
 https://landing.google.com/sre/book/chapters/addressing-cascading-failures.html
 
+## Escape Hatches
+
+It is not feasible or desirable for APIs like [`blob.Bucket`] to encompass the
+full functionality of every provider. Rather, we intend to provide a subset of
+the most commonly used functionality. There will be cases where a developer
+wants to access provider-specific functionality, which might consist of:
+
+1.  **Top-level APIs**. For example, `blob` does not expose a `Copy`, but some
+    provider might.
+1.  **Data fields**. For example, **blob** exposes a few attributes like
+    ContentType and Size, but S3 and GCS both have many more.
+1.  **Options**. Different providers may support different options for
+    functionality.
+
+Escape hatches provide the user a way to escape the Go Cloud abstraction to
```

Comment:
Since you are introducing a term, maybe emphasize it?

Reply:
Done.